internal/commands/server/k8s_e2e_test.go

```diff
@@ -299,7 +299,23 @@ func TestGatewayBasic(t *testing.T) {
 
 			require.Eventually(t, gatewayStatusCheck(ctx, resources, gatewayName, namespace, conditionReady), checkTimeout, checkInterval, "no gateway found in the allotted time")
 
-			err := resources.Delete(ctx, gw)
+			// check for ingress gateway config entry
+			var entries []api.ConfigEntry
+			var err error
+			require.Eventually(t, func() bool {
+				entries, _, err = client.ConfigEntries().List(api.IngressGateway, &api.QueryOptions{Namespace: e2e.ConsulNamespace(ctx)})
+				return err == nil && len(entries) == 1
+			}, 5*time.Minute, checkInterval, "config entry not created in allotted time")
+
+			// verify background re-sync of config entries after delete
+			_, err = client.ConfigEntries().Delete(api.IngressGateway, entries[0].GetName(), &api.WriteOptions{Namespace: e2e.ConsulNamespace(ctx)})
```

Comment:
this is probably good enough, but if you want to exercise a bit more of the re-sync you could try and delete the gateway service from Consul directly. Deleting the service should remove all of its config entries as well, and we should have the service get re-registered via the registration goroutine in the deployment and the config entries subsequently re-created when the service appears again in the catalog.

Reply:
This sounds intriguing! Will dig into this today

Reply:
After some digging, I discovered that deregistering the service from Consul does not remove config entries and intentions like we thought. Given that, I added additional assertions for both services and config entries so the tests are now more robust.